### PR TITLE
Add query string object parsing for makeUrl

### PR DESCRIPTION
**Pull request by tifa2UP**
### Summary
Added an optional utility that parses the query string parameters and appends it to the API call. E.g parses `{a: 1, b: 2}` to `?a=1&b=2`.

### Test Plan
Refactored the existing manual queryString usage to use the utility function, the API calls are formatted successfully.

### CODE REVIEW

File: src/utils/api.ts [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+
+  // The following section parses the query params for convenience
+  // E.g. parses {a: 1, b: 2} to "?a=1&b=2"
+  const parametersStringified = JSON.stringify(parameters);
+  const formattedParameters = JSON.parse(
+    parametersStringified,
+    (key, val) => (typeof val !== 'object' && val !== null ? String(val) : val), // Converts the value to a string in each key/value pair
+  );
+
+  const queryParameters = `?${new URLSearchParams(formattedParameters).toString()}`;
```
Comment (mmahalwy) [inline src/utils/api.ts:21]: does this work on SSR?

File: src/utils/api.ts [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
+ * @param parameters optional query params, {a: 1, b: 2} is parsed to "?a=1&b=2"
+ */
+export const makeUrl = (path: string, parameters?: Record<string, unknown>) => {
+  if (!parameters) {
+    return `${apiUrl}${path}`;
+  }
+
+  // The following section parses the query params for convenience
+  // E.g. parses {a: 1, b: 2} to "?a=1&b=2"
+  const parametersStringified = JSON.stringify(parameters);
```
Comment (mmahalwy) [inline src/utils/api.ts:15]: feels lame but i rather use a lib here like https://www.npmjs.com/package/qs or https://www.npmjs.com/package/query-string. They handle objects, arrays, etc etc

File: src/utils/api.ts (hunk 1 above)
Comment (tifa2UP) [reply src/utils/api.ts]: Yeah, next had a polyfill for it. It's just string parsing.

File: src/utils/api.ts (hunk 2 above)
Comment (tifa2UP) [reply src/utils/api.ts]: qs is `10kb`, and query-string is `32kb`. I added qs since we'll likely reuse it in the future. Thank you for the suggestion!

**Pull request merged**